index.html

```diff
@@ -1080,129 +1080,155 @@
+            instance with <code>bytes</code> set to <var>b</var>.
+            </li>
+          </ol>
+          <p>
+            The <dfn>data</dfn> attribute contains the data included in the <a>push message</a> when
+            included and the <a>user agent</a> verified its authenticity. The value will be set to
+            <code>null</code> in all other cases.
+          </p>
+        </section>
+        <section>
+          <h2>
+            Receiving a <a>Push Message</a>
+          </h2>
+          <p>
+            When the <a>user agent</a> receives a <a>push message</a> from the <a>push service</a>,
+            it MUST run the following steps.
+          </p>
+          <ol>
+            <li>Let <var>registration</var> be the <a>service worker registration</a> corresponding
+            to the <a>push message</a>.
+            </li>
+            <li>If <var>registration</var> is not found, abort these steps.
+            </li>
+            <li>Let <var>subscription</var> be the active <a>push subscription</a> for
+            <var>registration</var>.
+            </li>
+            <li>Initialize <var>data</var> to a value of <code>null</code>.
+            </li>
+            <li>If the <a>push message</a> contains a payload, perform the following steps:
+              <ol>
+                <li>Decrypt the <a>push message</a> using the private key from the key pair
+                associated with <var>subscription</var> and the process described in
+                [[!WEBPUSH-ENCRYPTION]]. This produces the plain text of the message.
+                </li>
+                <li>If the <a>push message</a> could not be decrypted for any reason, perform the
+                following steps:
+                  <ol>
+                    <li>Acknowledge the receipt of the <a>push message</a> according to
+                    [[!WEBPUSH-PROTOCOL]]. Though the message was not successfully received and
+                    processed, this prevents the push service from attempting to retransmit the
+                    message; a badly encrypted message is not recoverable.
+                    </li>
+                    <li>Discard the <a>push message</a>.
+                    </li>
+                    <li>Terminate this process.
+                    </li>
+                  </ol>A <code>push</code> event MUST NOT be fired for a <a>push message</a> that was
+                  not successfully decrypted using the key pair associated with the <a>push
+                  subscription</a>.
+                </li>
+                <li>Let <var>data</var> be a new <a>PushMessageData</a> instance with the decrypted
+                plain text of the <a>push message</a>.
+                </li>
+              </ol>
+            </li>
+            <li><a>Fire the push event</a> with <var>registration</var> and <var>data</var>, and
+            set <var>event</var> to the returned <a>trusted event</a>.
```

Comment:
^^^ I removed the `If the push message contains valid data...` phrasing here since `data` would be `null` if it doesn't.